Clean up hifiasm module + add bin file input

prototaxites opened the pull request:
- Tidy up hifiasm module code to have just a single script block, modifying inputs with variables
- Move `-o` option to directly before long_read input to ensure positional arguments are clearly separated from other arguments that can be lists of files
- Add optional input for .bin files to allow checkpointed resuming of a previous hifiasm run (e.g. to run in hi-c mode after running standard mode and reduce redundant computation)
- Reorganise output channels a bit
- Update nf-tests and add a test for resuming from bin files
- Add optional output channels for PAF and error-corrected FASTA, when requested

Questions:
- Is anyone using the separate .bin output channels? It might be cleaner to dump them out together but don't want to break anyone's workflow!

## PR checklist

Closes #XXX <!-- If this PR fixes an issue, please link it here! -->

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

prototaxites commented:
I'm not sure what to do about the empty file linting tests - I don't think there's enough data in the tests to produce non-empty files

SPPearce commented:
> I'm not sure what to do about the empty file linting tests - I don't think there's enough data in the tests to produce non-empty files

Typically you then replace the md5sum test with a file existence check.

prototaxites commented:
> Typically you then replace the md5sum test with a file existence check.

Is this standard practice for e.g. stub tests? I don't think I've ever written a test that way before.

SPPearce commented:
> > Typically you then replace the md5sum test with a file existence check.
> 
> Is this standard practice for e.g. stub tests? I don't think I've ever written a test that way before.

tests are allowed empty files provided they have "stub" in the name of the test

muffato commented on modules/nf-core/hifiasm/meta.yml:
```diff
@@ -117,55 +126,61 @@ output:
           description: File containing software versions
           pattern: "versions.yml"
+          ontologies: []
 authors:
```
Add yourself as an author. You deserved it ;)

fellen31 reviewed (approved):
> Is anyone using the separate .bin output channels? It might be cleaner to dump them out together but don't want to break anyone's workflow!

Would there be any other use case for the bin files than to resume the run? Thanks for updating the module to make that possible!

prototaxites commented:
> > Is anyone using the separate .bin output channels? It might be cleaner to dump them out together but don't want to break anyone's workflow!
> 
> Would there be any other use case for the bin files than to resume the run? Thanks for updating the module to make that possible!

I'm not aware of any other uses - you can dump some of the data out of them using hifiasm (e.g. `--write-ec` which will write the error-corrected reads bin file to fasta) - and am currently in the process of adding these outputs to the module. But I'm not aware that any other tool would use these outputs (and if you needed a specific one, you could still parse the output .bin file channel for the needed file).

prototaxites commented:
OK, I've added optional outputs for the error-corrected reads, and read overlaps in PAF format. I've also collapsed the bin files to a single output channel. I've also updated the documentation to add a bit more useful description about the outputs.

If these changes look good to folk I can go ahead and merge.

muffato commented on modules/nf-core/hifiasm/main.nf:
```diff
@@ -11,114 +11,100 @@ process HIFIASM {
+    tuple val(meta), path("${prefix}.{a_ctg,hic.a_ctg}.gfa")         , emit: alternate_contigs, optional: true
+    tuple val(meta), path("${prefix}.*.hap1.p_ctg.gfa")              , emit: hap1_contigs     , optional: true
+    tuple val(meta), path("${prefix}.*.hap2.p_ctg.gfa")              , emit: hap2_contigs     , optional: true
+    tuple val(meta), path("*.ec.fa.gz")                              , emit: fasta            , optional: true
+    tuple val(meta), path("*.ovlp.paf.gz")                           , emit: paf              , optional: true
```
Could you give names that tell about what's in files rather than their format ? e.g. `corrected_reads` for `fasta`, and `read_overlap` for `paf`.
